source/distributed/tria_base.cc

```diff
@@ -163,6 +166,32 @@ namespace parallel
         return;
       }
 
+
+    {
+      // find ghost owners
+      for (typename Triangulation<dim,spacedim>::active_cell_iterator
+           cell = this->begin_active();
+           cell != this->end();
+           ++cell)
+        if (cell->is_ghost())
+          number_cache.ghost_owners.insert(cell->subdomain_id());
+
+      Assert(number_cache.ghost_owners.size() < Utilities::MPI::n_mpi_processes(mpi_communicator), ExcInternalError());
+    }
+
+    {
+      // find level ghost owners
+      for (typename Triangulation<dim,spacedim>::cell_iterator
```

Comment:
Problem: I don't know if this is the case, because we only do this in ```p::d::Triangulation``` and this code currently sits in the base class ```p::Triangulation```. I am moving stuff around...